### PR TITLE
update openSUSE data with latest versions and clean up really old ones

### PR DESCRIPTION
**Pull request by wrosenauer**

### CODE REVIEW

**Comment by glandium** [comment]
"Merged" in e13a0fa11a48f3302c9033ffb8188be089881ee8.

File: distro_data.json [hunk 1]
```diff
@@ -351,6 +323,26 @@
             "stdc++": "5.3.1"
         }
     },
+    "openSUSE Leap 42.3": {
+        "release": {
+            "date": "2017-07",
+            "eol": "2019-012"
```
Comment (glandium) [inline distro_data.json]: Noticed the typo here just now. Is it actually 01 or 12?

Comment (glandium) [reply distro_data.json]: 01 according to https://en.opensuse.org/Lifetime

Comment (wrosenauer) [reply distro_data.json]: confirmed, sorry for the typo

**Pull request closed without merging**